### PR TITLE
removes deprecated errorConsumer

### PR DESCRIPTION
**Pull request by OlegDokuka**
Signed-off-by: Oleh Dokuka <shadowgun@i.ua>

### CODE REVIEW

File: rsocket-core/src/main/java/io/rsocket/RSocketFactory.java [hunk 1]
```diff
@@ -395,7 +395,6 @@ public ClientRSocketFactory fragment(int mtu) {
 
     /** @deprecated this is deprecated with no replacement. */
     public ClientRSocketFactory errorConsumer(Consumer<Throwable> errorConsumer) {
-      connector.errorConsumer(errorConsumer);
       return this;
     }
```
Comment (rstoyanchev) [inline rsocket-core/src/main/java/io/rsocket/RSocketFactory.java:399]: Let's mention in the Javadoc that as of 1.0 this method is a no-op.

File: rsocket-core/src/main/java/io/rsocket/RSocketFactory.java [hunk 2]
```diff
@@ -499,7 +498,6 @@ public ServerRSocketFactory fragment(int mtu) {
 
     /** @deprecated this is deprecated with no replacement. */
     public ServerRSocketFactory errorConsumer(Consumer<Throwable> errorConsumer) {
-      server.errorConsumer(errorConsumer);
       return this;
     }
```
Comment (rstoyanchev) [inline rsocket-core/src/main/java/io/rsocket/RSocketFactory.java:502]: Same here, mention the method is a no-op.

File: rsocket-core/src/main/java/io/rsocket/core/RSocketRequester.java [hunk 3]
```diff
@@ -610,9 +611,9 @@ private void handleStreamZero(FrameType type, ByteBuf frame) {
         break;
       default:
         // Ignore unknown frames. Throwing an error will close the socket.
-        errorConsumer.accept(
-            new IllegalStateException(
-                "Client received supported frame on stream 0: " + frame.toString()));
+        if (LOGGER.isDebugEnabled()) {
+          LOGGER.debug("Requester received unsupported frame on stream 0: " + frame.toString());
+        }
```
Comment (rstoyanchev) [inline rsocket-core/src/main/java/io/rsocket/core/RSocketRequester.java:616]: I wonder if this shouldn't be visible in production logs, perhaps `INFO` level?

Comment (rstoyanchev) [inline rsocket-core/src/main/java/io/rsocket/core/RSocketRequester.java:616]: I wonder if this shouldn't be visible in production logs, perhaps `INFO` level?

Comment (OlegDokuka) [reply rsocket-core/src/main/java/io/rsocket/core/RSocketRequester.java:616]: Interesting question. This is non-harmful thing that might happen. In general, it means that there is something wrong but it does not break the general work

File: rsocket-core/src/main/java/io/rsocket/RSocketFactory.java (hunk 2 above)
Comment (OlegDokuka) [reply rsocket-core/src/main/java/io/rsocket/RSocketFactory.java:512]: fixed

File: rsocket-core/src/main/java/io/rsocket/RSocketFactory.java (hunk 1 above)
Comment (OlegDokuka) [reply rsocket-core/src/main/java/io/rsocket/RSocketFactory.java:404]: fixed

**Pull request merged**